ch_pipeline/processing/base.py

```diff
@@ -157,11 +159,20 @@ def ls(self):
         if not base.exists():
             raise ValueError("Base path %s does not exist." % base)
 
-        file_regex = re.compile("^%s$" % self.tag_pattern)
+        file_regex = re.compile(f"^{self.tag_pattern}$")
 
-        entries = [path.name for path in base.glob("*") if file_regex.match(path.name)]
+        entries = [
```

Comment:
I think it would be good to move the scanning for `mtime`'s into the `time_sort` conditional. Stat-ing files can be pretty slow on cedar and so it's best to avoid it if you can.

Reply:
Ah, good call. I thought I had done that, but I guess I just thought about it...